client/my-sites/posts/main.jsx

```diff
@@ -116,13 +116,14 @@ const PostsMain = React.createClass( {
 
 } );
 
-export default connect( ( state ) => {
+export default connect( ( state, props ) => {
 	const siteId = getSelectedSiteId( state );
 	const draftsQuery = {
 		type: 'post',
 		status: 'draft',
 		number: 6,
-		order_by: 'modified'
+		order_by: 'modified',
+		author: props.author
```

Comment:
Kinda minor, but I don't think `PostQueryManager` is smart enough to consider `null` the default author value ([reference](https://github.com/Automattic/wp-calypso/blob/7bc68d6bd3d39d7ad8df3273cfe6ddca9696f578/client/lib/query-manager/post/key.js#L12-L22), [as passed by posts controller](https://github.com/Automattic/wp-calypso/blob/7bc68d6bd3d39d7ad8df3273cfe6ddca9696f578/client/my-sites/posts/controller.js#L28)), so it'll treat this and a query lacking the `author` property as distinct. Maybe a problem we should solve on the `PostQueryManager` side though (`null` might have special meaning for certain fields, which is why I didn't check for this).